crypto/http/http_lib.c

```diff
@@ -114,3 +120,49 @@ int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
     OPENSSL_free(buf);
     return 0;
 }
+
+int http_use_proxy(const char *no_proxy, const char *server)
+{
+    size_t sl;
+    const char *found = NULL;
+
+    if (!ossl_assert(server != NULL))
+        return 0;
+    sl = strlen(server);
+
+    if (no_proxy == NULL)
+        no_proxy = getenv("no_proxy");
+    if (no_proxy == NULL)
+        no_proxy = getenv("NO_PROXY");
```

Comment:
Should this be `OPENSSL_NO_PROXY` or similar? Or perhaps better `OPENSSL_NO_HTTPS_PROXY`?

Not sure why we have both case variants. We don't do that for any other env variable AFAIK.

Reply:
Good point to add `OPENSSL_NO_PROXY`, done.

Regarding the lower-case variants, I found that some projects do so for compatibility - I can kick them out here if you prefer.

Reply:
It seems that there is no normative standard on the proxy environment variables.

Here is a nice description of what Git does: https://github.com/request/request#controlling-proxy-behaviour-using-environment-variables
and this PR follows that model.

Curl is weird in the sense that it only supports the lower_case `http_proxy` but both lower- and upper-case `HTTPS_PROXY` - see https://ec.haxx.se/usingcurl/usingcurl-proxies

BTW, apparently no implementation uses `NO_HTTPS_PROXY` or the like - the only env vars typically used for exclusion are `NO_PROXY` and/or `no_proxy` - these are used for both HTTP and HTTPS.

https://en.wikibooks.org/wiki/Guide_to_Unix/Environment_Variables writes

> The names of environment variables are case-sensitive; by convention they are uppercase.

Unfortunately it is not guaranteed that `getenv` is case insensitive - see,  e.g., https://www.php.net/manual/en/function.getenv.php